hbase-server/src/test/java/org/apache/hadoop/hbase/master/procedure/TestSnapshotProcedureConcurrently.java

```diff
@@ -136,4 +138,48 @@ public void run() {
     SnapshotTestingUtils.confirmSnapshotValid(TEST_UTIL, snapshotProto, TABLE_NAME, CF);
     SnapshotTestingUtils.confirmSnapshotValid(TEST_UTIL, snapshotOnSameTableProto, TABLE_NAME, CF);
   }
+
+  @Test
+  public void testItCanEnableTableWhileSnapshotProcedureIsRunning() throws Exception {
```

Comment:
This test fails without my changes, both procedures will run forever. This also functions as the reproduction of the phenomenon. 

For reproducibility, I manually set the table state to ENABLING, and then kick off the EnableTableProcedure at the stage it gets stuck on (after subprocesses finish)